CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+  URL "https://github.com/ruslo/hunter/archive/v0.12.44.tar.gz"
+  SHA1 "54963db4cd6e2a8780643a3cc48bdbbefedec6cc"
+)
+
+hunter_add_package(ZLIB)
```

Comment:
You are right, but protocol buffers CMake scripts doesn't find sources if you use them from different root then `.cmake/`. I made a modification (`get_filename_component(protobuf_source_dir ${protobuf_SOURCE_DIR} PATH)` becomes a simple set operation), but install stuff doesn't work anyway. So I made the following changes. The project line has been duplicated. In this case protobuf_SOURCE_DIR remains the old, the cmake/CMakeLists.txt file can be used as in the past but you can compile with root CMakeLists.txt.

Probably that also works that we override protobuf_SOURCE_DIR variable at the beginning of cmake/CMakeLists.txt